File: android/src/main/kotlin/com/techlabs/flutter/flutter_techlabs_library/lib/model/NoticeCheckResult.kt
package com.techlabs.flutter.flutter_techlabs_library.lib.model

/**
 * 공지사항 체크 결과를 나타내는 sealed class.
 */
sealed class NoticeCheckResult {
    data class HasNewNotice(val latestIndex: Int) : NoticeCheckResult()
    data object NoNewNotice : NoticeCheckResult()
    data object Error : NoticeCheckResult()
}

File: lib/src/models/notice_check_result.dart
sealed class NoticeCheckResult {
  const NoticeCheckResult();
}

final class HasNewNotice extends NoticeCheckResult {
  final int latestIndex;

  const HasNewNotice(this.latestIndex);
}

final class NoNewNotice extends NoticeCheckResult {
  const NoNewNotice();
}

final class NoticeCheckError extends NoticeCheckResult {
  const NoticeCheckError();
}
